proto/client/gClient.go

```diff
@@ -144,76 +139,66 @@ func newClientConnection(host string) (conn *grpc.ClientConn, err error) {
-			continue
+	status := confClient.Conn.GetState()
+	if status == connectivity.Ready {
+		logger.GrpcLog.Debugln("connectivity ready")
+		rreq := &protos.NetworkSliceRequest{RestartCounter: selfRestartCounter, ClientId: myid, MetadataRequested: confClient.MetadataRequested}
+		if stream, err = confClient.Client.NetworkSliceSubscribe(context.Background(), rreq); err != nil {
+			return stream, fmt.Errorf("failed to subscribe: %v", err)
 		}
+		return stream, nil
+	} else if status == connectivity.Idle {
+		return nil, fmt.Errorf("connectivity status idle")
+	} else {
+		return nil, fmt.Errorf("connectivity status idle")
+	}
+}
 
-		logger.GrpcLog.Infoln("stream msg received")
-		logger.GrpcLog.Debugf("network slices %d, RC of configpod %d", len(rsp.NetworkSlice), rsp.RestartCounter)
-		if configPodRestartCounter == 0 || (configPodRestartCounter == rsp.RestartCounter) {
-			// first time connection or config update
-			configPodRestartCounter = rsp.RestartCounter
-			if len(rsp.NetworkSlice) > 0 {
-				// always carries full config copy
-				logger.GrpcLog.Infoln("first time config received", rsp)
-				commChan <- rsp
-			} else if rsp.ConfigUpdated == 1 {
-				// config delete , all slices deleted
-				logger.GrpcLog.Infoln("complete config deleted")
-				commChan <- rsp
-			}
-		} else if len(rsp.NetworkSlice) > 0 {
-			logger.GrpcLog.Errorln("config received after config pod restart")
-			// config received after config pod restart
-			configPodRestartCounter = rsp.RestartCounter
+// subscribeToConfigPod subscribing channel to ConfigPod to receive configuration
+// using stream and communication channel as inputs
+func (confClient *ConfigClient) subscribeToConfigPod(commChan chan *protos.NetworkSliceResponse, stream protos.ConfigService_NetworkSliceSubscribeClient) {
+	rsp, err := stream.Recv()
+	if err != nil {
+		logger.GrpcLog.Errorf("failed to receive message from stream: %v", err)
+		return
+	}
+
+	logger.GrpcLog.Infoln("stream message received")
+	logger.GrpcLog.Debugf("network slices %d, RC of config pod %d", len(rsp.NetworkSlice), rsp.RestartCounter)
+	if configPodRestartCounter == 0 || (configPodRestartCounter == rsp.RestartCounter) {
+		// first time connection or config update
+		configPodRestartCounter = rsp.RestartCounter
+		if len(rsp.NetworkSlice) > 0 {
+			// always carries full config copy
+			logger.GrpcLog.Infoln("first time config received", rsp)
+			commChan <- rsp
+		} else if rsp.ConfigUpdated == 1 {
+			// config delete , all slices deleted
+			logger.GrpcLog.Infoln("complete config deleted")
 			commChan <- rsp
-		} else {
-			logger.GrpcLog.Errorln("config pod is restarted and no config received")
 		}
+	} else if len(rsp.NetworkSlice) > 0 {
+		logger.GrpcLog.Errorln("config received after config pod restart")
+		// config received after config pod restart
+		configPodRestartCounter = rsp.RestartCounter
+		commChan <- rsp
+	} else {
+		logger.GrpcLog.Errorln("config pod is restarted and no config received")
```

Comment:
Have you tested these changes with the [nrf#132](https://github.com/omec-project/nrf/pull/132)? I am asking this because I do not see any logging about any of these lines been executed. For example, the `nrf` (before your patch/PR) shows the log below but with your PR, I do not see that line. Moreover, `gnbsim` test fails for all profiles
```
2024-10-17T16:12:27.887Z	INFO	client/gClient.go:203	first time config received RestartCounter:2637084572 NetworkSlice:{Name:"default" Nssai:{Sst:"1" Sd:"010203"} DeviceGroup:{Name:"5g-gnbsim-user-group1" IpDomainDetails:{Name:"pool1" DnnName:"internet" UePool:"172.250.1.0/16" DnsPrimary:"8.8.8.8" Mtu:1460 UeDnnQos:{DnnMbrUplink:1000000000 DnnMbrDownlink:1000000000 TrafficClass:{Name:"platinum" Qci:9 Arp:6 Pdb:300 Pelr:6}}} Imsi:"208930100007500" Imsi:"208930100007501" Imsi:"208930100007502" Imsi:"208930100007503" Imsi:"208930100007504" Imsi:"208930100007505" Imsi:"208930100007506" Imsi:"208930100007507"} Site:{SiteName:"enterprise" Gnb:{Name:"gnb1" Tac:1} Gnb:{Name:"gnb2" Tac:2} Plmn:{mcc:"208" mnc:"93"} Upf:{UpfName:"upf"}} AppFilters:{PccRuleBase:{FlowInfos:{FlowDesc:"permit out ip from any to assigned" TosTrafficClass:"IPV4" FlowDir:BIDIRECTIONAL FlowStatus:ENABLED} RuleId:"ALLOW-ALL" Qos:{Var5qi:9 Arp:{PL:6 PC:MAY_PREEMPT PV:PREEMPTABLE}} Priority:250}}} ConfigUpdated:1	{"component": "Config5g", "category": "GRPC"}
```